hera_sim/visibilities/vis_cpu.py

```diff
@@ -45,22 +45,35 @@ class VisCPU(VisibilitySimulator):
         Whether to use the GPU version of vis_cpu or not. Default: False.
     mpi_comm : MPI communicator
         MPI communicator, for parallelization.
-    **kwargs
-        Passed through to :class:`~.simulators.VisibilitySimulator`.
+    ref_time
+        A reference time for computing adjustments to the co-ordinate transforms using
+        astropy. For best fidelity, set this to a mid-point of your observation times.
+        By default, if `correct_source_positions` is True, will use the first
+        observation time.
+    correct_source_positions
+        Whether to correct the source positions using astropy and the reference time.
+        Default is True if `ref_time` is given otherwise False.
     """
 
+    conjugation_convention = "ant1<ant2"
+    time_ordering = "time"
+
+    diffuse_ability = False
+    __version__ = __version__
+
     def __init__(
         self,
-        bm_pix=100,
-        use_pixel_beams=True,
-        polarized=False,
-        precision=1,
-        use_gpu=False,
+        bm_pix: int = 100,
+        use_pixel_beams: bool = True,
+        polarized: bool = False,
+        precision: int = 1,
+        use_gpu: bool = False,
         mpi_comm=None,
-        **kwargs
+        ref_time: Optional[Union[str, Time]] = None,
```

Comment:
From the doc https://docs.python.org/3.10/library/typing.html#typing.Optional

`Optional[X]` is equivalent to `X | None` (or `Union[X, None]`).

So `ref_time: Optional[Union[str, Time]] = None` means `ref_time` can be type `Union[str, Time, None]`. The `Optional` qualifier seems irrelevant here.

Reply:
I don't see how it's irrelevant? One of the options is None, so it definitely needs the `Optional`?

Reply:
In case you're still confused, I think that the value of the `Optional` type is its readability. Typing something as `Union[..., None]` doesn't make it quite as clear that the parameter is optional (i.e. doesn't need to be specified when calling the method/function) as the `Optional[...]` typing makes it.

Also, setting the default parameter value to `None` doesn't always mean that it's an optional parameter. Maybe it's not best practice to do this, but I've used it at least once in `hera_sim`: the `foregrounds.DiffuseForeground.__call__` method has the `Tsky` default set to `None`, and it will raise an error if it is not specified.

Reply:
Yes, I see now that its replace the Numpy docstring "optional" description.